source/MaterialXContrib/MaterialXNode/MaterialXNode.cpp

```diff
@@ -120,59 +120,66 @@ bool MaterialXNode::getInternalValue(const MPlug& plug, MDataHandle& dataHandle)
     return true;
 }
 
-bool MaterialXNode::setInternalValue(const MPlug& plug, const MDataHandle& dataHandle)
+void MaterialXNode::createAndRegisterFragment(bool reloadDocument)
```

Comment:
Could you please remove this argument and instead just reset _document to nullptr before calling this method? This is a good idea anyway because the previous document should be destroyed before constructing the new one, or else both of them would be in memory at the same time for no reason.